### PR TITLE
Feature/policy term period certificates

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/web/InternalController.java [hunk 1]
```diff
@@ -195,8 +199,9 @@ public ProductSearchResultDTO searchPaged(ProductSearchRequestDTO searchRequest)
 
     List<TraceProductDTO> traces = traceProductService.getTracesByProductId(product.get().id);
 
-    if (product.get().certificateKey != null && product.get().certificateBucket != null) {
-      return ok(new InsuranceStatusDTO(product.get(), generateCertificateURL(product.get()), traces));
+    final InsuranceCertificate insuranceCertificate = insuranceCertificateFactory.buildCertificate(product.get(), Instant.now());
+    if (insuranceCertificate.isGenerated()) {
```
[the remark this answers is not in the document]
Comment (palmenhq) [reply src/main/java/com/hedvig/productPricing/service/web/InternalController.java]: No. (irl discussion)

File: src/main/java/com/hedvig/productPricing/service/serviceIntegration/botService/dto/Address.java [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
 import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 
 @Embeddable
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
+@EqualsAndHashCode
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/serviceIntegration/botService/dto/Address.java]: You can remove this, since you have the `@Data`

Comment (palmenhq) [reply src/main/java/com/hedvig/productPricing/service/serviceIntegration/botService/dto/Address.java]: oh interesting. Unsure how it ended up here 🤷‍♀️

File: src/main/java/com/hedvig/productPricing/util/VelocityTools.java [hunk 3]
```diff
@@ -24,12 +24,30 @@ public boolean isRental(ProductEntity ent) {
...
   }
 
+  public boolean isRental(final InsuranceCertificate insuranceCertificate) {
+    return
+      insuranceCertificate.getProductType() == Product.ProductTypes.RENT
+        || insuranceCertificate.getProductType() == Product.ProductTypes.STUDENT_RENT
+        || insuranceCertificate.getProductType() == Product.ProductTypes.SUBLET_RENTAL;
+  }
+
   public boolean isStudent(ProductEntity ent) {
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/util/VelocityTools.java]: Minor comment, but should we mark those methods as deprecated?

Comment (palmenhq) [reply src/main/java/com/hedvig/productPricing/util/VelocityTools.java]: i think they're still used?

**Pull request merged**